앱스토어 업데이트 알림창 설정 변경

Eunice0927 opened the pull request:
<!-- 작업 동기에는 해당 작업을 수행한 이유 또는 목적을 간단히 적어주세요. -->
<!-- in progress는 진행중인 연관 이슈, close에는 닫고싶은 issue를 적습니다. -->
## 개요
- in progress
- close #198 
- 작업 요약: 앱스토어 업데이트 알림창 설정 변경

<!-- PR의 핵심이 되는 작업 내용을 적어주세요. -->
## 변경 사항
- 기존: 앱스토어와 현재 설치 버전이 불일치하면 업데이트 알림
- 수정: 앱스토어 버전보다 현재 설치 버전이 더 작으면 업데이트 알림

<!-- 노션 팀 페이지에 정리한 글의 링크나 참고한 블로그의 링크를 남겨주세요. -->
## 공유사항(배운 것, 참고하면 좋을 것)
- [[iOS] Swift 강제 업데이트 Alert 설정](https://velog.io/@chaentopia/iOS-Swift-%EA%B0%95%EC%A0%9C-%EC%97%85%EB%8D%B0%EC%9D%B4%ED%8A%B8-Alert-%EC%84%A4%EC%A0%95)

<!-- 이미지 가로 크기 216 고정 -->
## 스크린샷
|제목|작업 전|작업 후|
|:-:|:-:|:-:| 
||||

<!-- 질문, 중점적으로 확인받고 싶은 내용 또는 주의사항 등 자유로운 전달사항 적어주세요. -->
## 전달사항
-

Eunice0927 committed:
Feat: 앱스토어 버전보다 기기 설치 버전이 낮으면 업데이트 알림창이 뜨도록 수정

Eunice0927 committed:
Fix: 불필요한 import 삭제 및 수정

qlrmr111 reviewed (approved):
수고하셨습니당
근데 저희 print 안쓰고 Logger쓰기로 했지 않나요?

Eunice0927 committed:
Fix: print 대신 logger를 사용하도록 수정

Eunice0927 commented:
> 수고하셨습니당 근데 저희 print 안쓰고 Logger쓰기로 했지 않나요?

꼼꼼한 확인 감사드려요!
수정해서 커밋 제출했습니다~ :)

Eunice0927 committed:
Fix: 스토어 버전보다 현재 버전이 더 클 경우에 대응

앱스토어 버전과 상기 url에서 받아오는 데이터 버전의 업데이트 시차가 약 24시간 존재하여, 버전이 같은 경우와 url data의 버전이 더 작은 경우는 업데이트를 띄우지 않도록 설정

Eunice0927 committed:
Remove: 불필요한 주석 삭제

Eunice0927 committed:
Style: 불필요한 줄바꿈 삭제

Eunice0927 committed:
Style: AuthManager에서 사용하지 않는 코드 삭제

Eunice0927 committed:
Refactor: 기존의 코드를 한 줄로 개선

Eunice0927 commented on PJ3T3_Postie/Manager/AppStoreUpdateChecker.swift:
```diff
@@ -34,31 +34,32 @@ struct AppStoreUpdateChecker {
+                Logger.version.info("----> 현재 버전 셋째자리: \(splitCurrentVersion[2])")
                 return true
             }
             
```
안녕하세요 재성님!

코드 꼼꼼히 리뷰 해 주셔서 감사드립니다!! 🥹

앱스토어 버전과 상기 url에서 받아오는 데이터 버전의 업데이트 시차가 약 24시간 존재하여, 이미 업데이트 한 사람에게도하루동안 업데이트 알림이 뜨는 이슈를 해결하던 중 로직에 오류가 있었네요..!
말씀 해 주신 버전이 같은 경우와 url에서 받은 data의 버전이 더 작은 경우는 업데이트를 띄우지 않도록 코드 작성 해 보았습니다.
혹시 추가로 더 개선이 필요해 보이는 부분이 있다면 말씀 부탁드립니다! 😊

감사합니다.
